packages/frontend/src/redux/slices/staking/index.js

```diff
@@ -1,4 +1,52 @@
+import { createSelector } from "reselect";
+
+import createParameterSelector from "../createParameterSelector";
+
 const SLICE_NAME = 'staking';
 
+const getAccountIdParam = createParameterSelector((params) => params.accountId);
+const getValidatorIdParam = createParameterSelector((params) => params.validatorId);
+
 // Top level selectors
-export const selectStakingSlice = (state) => state[SLICE_NAME];
+export const selectStakingSlice = (state) => state[SLICE_NAME] || {};
+
+// accounts - state
+export const selectStakingAccounts = createSelector(selectStakingSlice, (staking) => staking.accounts || []);
+
+export const selectStakingCurrentAccountbyAccountId = createSelector(
+    [selectStakingAccounts, getAccountIdParam], 
+    (accounts, accountId) => accounts.find((account) => account.accountId === accountId)
+);
+
+export const selectStakingAccountsFirst = createSelector(selectStakingAccounts, (accounts) => accounts[0] || {});
+
+export const selectStakingAccountsSecond = createSelector(selectStakingAccounts, (accounts) => accounts[1] || {});
+
+// lockup - state
+export const selectStakingLockup = createSelector(selectStakingSlice, (staking) => staking.lockup || {});
+
+export const selectStakingLockupId = createSelector(selectStakingLockup, (lockup) => lockup.lockupId || '');
+
+export const selectStakingContract = createSelector(selectStakingLockup, (lockup) => lockup.contract || {});
+
+// allValidators - state
+export const selectStakingAllValidators = createSelector(selectStakingSlice, (staking) => staking.allValidators || []);
+
+export const selectStakingAllValidatorsLength = createSelector(selectStakingAllValidators, (allValidators) => allValidators.length);
+
+export const selectStakingFindContractByValidatorId = createSelector(
+    [selectStakingAllValidators, getValidatorIdParam],
+    (allValidators, validatorId) => allValidators.find((validator) => validator.accountId === validatorId)?.contract || {}
+);
+
+// accountsObj - state
+export const selectStakingAccountsObj = createSelector(selectStakingSlice, (staking) => staking.accountsObj || {});
+
+export const selectStakingAccountsObjAccountId = createSelector(selectStakingAccountsObj, (accountsObj) => accountsObj.accountId || '');
```

Comment:
As someone new to this codebase, the `accountsObj` state property object is a little confusing 😄 . Is there possibly a more descriptive name for what these selectors are returning that distinguishes it from the other selectors in this module?

I'm interested to hear your thoughts - it may very well make more sense to leave it as is for the time being. I just like to think about renaming when writing selectors because it can provide opportunities to make things more clear from the caller's perspective.

Reply:
Agree, you're right. I renamed it into `selectStakingMainAccountId` and `selectStakingLockupAccountId`. 

The next step will be to finish refactoring staking actions/reducer into the redux toolkit version, I will definitely improve some of this functionality. And next, we are planning to refactor the whole redux state tree - it's a bit of a mess right now, because the project was expanding so fast, and there never was enough time to take care of it. So then I will definitely refactor the whole staking state tree.